pythran/pythonic/numpy/reduce.hpp

```diff
@@ -20,53 +20,61 @@ namespace pythonic
 
   namespace numpy
   {
+    template <class Op, size_t N, bool vector_form>
+    struct _reduce {
+      template <class E, class F>
+      F operator()(E e, F acc)
```

Comment:
On Wed, Sep 02, 2015 at 11:29:49AM -0700, pbrunet wrote:
> > @@ -20,53 +20,61 @@ namespace pythonic
> >  
> >    namespace numpy
> >    {
> > +    template <class Op, size_t N, bool vector_form>
> > +    struct _reduce {
> > +      template <class E, class F>
> > +      F operator()(E e, F acc)
> 
> Why do we want a copy?
To enforce return by value and pass by value which makes it easier to
optimize by the compiler?

I recall a conversation with @joker-eph on the subject :-)